XXX bring glibmm268's Glib::ustring and range-v3 together

backport pending "Use std::invoke_result_t not ::result_of to fix C++20 tdesktop 4.3.4"
https://github.com/telegramdesktop/tdesktop/pull/25527

Index: Telegram/SourceFiles/platform/linux/specific_linux.cpp
--- Telegram/SourceFiles/platform/linux/specific_linux.cpp.orig
+++ Telegram/SourceFiles/platform/linux/specific_linux.cpp
@@ -243,12 +243,16 @@ void LaunchGApplication() {
 	const auto activatableNames = [&] {
 		try {
 			if (connection) {
-				return ListActivatableNames(connection);
+				const auto GlibNames =
+				    ListActivatableNames(connection);
+				return std::vector<std::string>(
+				    GlibNames.begin(), GlibNames.end()
+				);
 			}
 		} catch (...) {
 		}
 
-		return std::vector<Glib::ustring>();
+		return std::vector<std::string>();
 	}();
 
 	const auto freedesktopNotifications = [&] {
@@ -346,9 +350,10 @@ void LaunchGApplication() {
 
 			using Window::Notifications::Manager;
 			using NotificationId = Manager::NotificationId;
-			using NotificationIdTuple = std::result_of<
-				decltype(&NotificationId::toTuple)(NotificationId*)
-			>::type;
+			using NotificationIdTuple = std::invoke_result_t<
+				decltype(&NotificationId::toTuple),
+				NotificationId*
+			>;
 
 			const auto notificationIdVariantType = [] {
 				try {
